portindex2json/portindex2json.tcl

```diff
@@ -64,31 +64,82 @@ proc remove_extra_braces {text} {
+                }
+            } elseif {$arg eq "-help" || $arg eq "-h" || $arg eq "--help" } {
+                puts stdout "\nUsage: /path/to/tclsh portindex2json.tcl <path-to-portindex> -info key=vlaue"
+                puts stdout "\nEach '-info key=value' combinaton adds one key-value pair to the JSON output inside the parent key: \"info\""
+                puts stdout "\nYou may provide any number of '-info key=value' combinations."
+                puts stdout "\nThe arguments <path-to-portindex> and `-info key=value` are position independent.\n"
+                exit 0
+            } else {
+                puts stderr "\nError: $arg is an invalid argument. Expecting -info, -help.\n"
+                exit 1
+            }
+        }
+        default {
+            incr filename
+            if { [catch {open [lindex $argv $i] r} fid] } {
+                puts stderr "\nError: Could not open [lindex $argv $i]. Make sure the path to the PortIndex file is correct\n"
+                exit 1
+            } else {
+                set fd [open [lindex $argv $i] r]
+            }
         }
     }
-    lappend objects [::json::write object {*}[array get json_portinfo]]
 }
 
-chan configure stdout -encoding utf-8
-puts [::json::write array {*}$objects]
+if {[info exists fd] && $filename eq 1} {
+    chan configure $fd -encoding utf-8
+    while {[gets $fd line] >= 0} {
+        if {[llength $line] != 2} {
+            continue
+        }
+        set len [lindex $line 1]
+        set line [read $fd $len]
+        array unset portinfo
+        array set portinfo $line
+        array unset json_portinfo
+        foreach key [array names portinfo] {
+            if {$key == "categories" || $key == "variants" || [string match "depends_*" $key]} {
+                set json_portinfo($key) [tcl2json_list $portinfo($key)]
+            } elseif {$key == "maintainers"} {
+                set json_portinfo($key) [parse_maintainers $portinfo($key)]
+                set json_portinfo(closedmaintainer) [is_closedmaintainer $portinfo($key)]
+            } elseif {$key == "long_description"} {
+                set json_portinfo($key) [::json::write string [remove_extra_braces $portinfo($key)]]
+            } else {
+                set json_portinfo($key) [::json::write string $portinfo($key)]
+            }
+        }
+        lappend objects [::json::write object {*}[array get json_portinfo]]
+    }
+
+    if {[array exists json_info]} {
+        set json_output(info) [::json::write object {*}[array get json_info]]
+    }
+
+    set json_output(ports) [::json::write array {*}$objects]
+
+    chan configure stdout -encoding utf-8
+    puts [::json::write object {*}[array get json_output]]
+} elseif {$filename eq 0}  {
+    puts stderr "\nError: The path to the PortIndex file was not provided. Use -help.\n"
+    exit 1
+} else {
+    puts stderr "\nError: You provided invalid or more than one paths to the PortIndex file. Use -help.\n"
```

Comment:
You should catch "more than one paths" already when parsing the arguments, and don't even try to check for file existence of the second non-optional parameter.

Reply:
With the way arguments are parsed currently, the recent changes do not something exactly to this, but I tried to keep it close.